main_test.go

```diff
@@ -1156,6 +1167,83 @@ func TestPuts(t *testing.T) {
 		path := t.TempDir()
 		transformer := "prefix=" + path + ":" + remotePath
 
+		Convey("Invalid metadata throws an error", func() {
+			file1 := filepath.Join(path, "file1")
+
+			internal.CreateTestFile(t, file1, "some data1")
+
+			setName := "invalidMetadataTest1"
+			setMetadata := "testKey:testValue:anotherValue"
+
+			exitCode, err := s.runBinary(t, "add", "--name", setName, "--transformer", transformer,
+				"--path", path, "--metadata", setMetadata)
+
+			So(exitCode, ShouldEqual, 1)
+			So(err, ShouldContainSubstring, "invalid meta: testKey:testValue:anotherValue")
+
+			setName = "invalidMetadataTest2"
+			setMetadata = "ibackup:set:invalidMetadataTest2"
+
+			exitCode, _ = s.runBinary(t, "add", "--name", setName, "--transformer", transformer,
+				"--path", path, "--metadata", setMetadata)
+
+			So(exitCode, ShouldEqual, 1)
+
+			setName = "invalidMetadataTest3"
+			setMetadata = "name:name:name:name"
+
+			exitCode, _ = s.runBinary(t, "add", "--name", setName, "--transformer", transformer,
+				"--path", path, "--metadata", setMetadata)
+
+			So(exitCode, ShouldEqual, 1)
+		})
+
+		Convey("Putting metadata on a set adds that metadata to every file in the set", func() {
+			attributePrefix := "attribute: user:"
+			valuePrefix := "\nvalue: "
+
+			file1 := filepath.Join(path, "file1")
+			file2 := filepath.Join(path, "file2")
+			file3 := filepath.Join(path, "file3")
+
+			internal.CreateTestFile(t, file1, "some data1")
+			internal.CreateTestFile(t, file2, "some data2")
+			internal.CreateTestFile(t, file3, "some data3")
+
+			setName := "metadataTest"
+			fileNames := []string{"file1", "file2", "file3"}
+			setMetadata := "testKey1:testValue1;testKey2:testValue2"
+
+			s.addSetForTestingWithMetadata(t, setName, transformer, path, setMetadata)
+
+			s.waitForStatus(setName, "\nStatus: complete", 60*time.Second)
+
+			for _, fileName := range fileNames {
+				output := getRemoteMeta(filepath.Join(remotePath, fileName))
+				So(output, ShouldContainSubstring, attributePrefix+"testKey1\n")
+				So(output, ShouldContainSubstring, valuePrefix+"testValue1\n")
+				So(output, ShouldContainSubstring, attributePrefix+"testKey2\n")
+				So(output, ShouldContainSubstring, valuePrefix+"testValue2\n")
+			}
+
+			newName := setName + ".v2"
+			setMetadata = "testKey2:testValue2Updated"
+
+			s.addSetForTestingWithMetadata(t, newName, transformer, path, setMetadata)
+
+			s.waitForStatus(newName, "\nStatus: complete", 60*time.Second)
+
+			for _, fileName := range fileNames {
+				output := getRemoteMeta(filepath.Join(remotePath, fileName))
+				So(output, ShouldContainSubstring, attributePrefix+"testKey1\n")
+				So(output, ShouldContainSubstring, valuePrefix+"testValue1\n")
+				So(output, ShouldContainSubstring, attributePrefix+"testKey2\n")
+				So(output, ShouldContainSubstring, valuePrefix+"testValue2Updated\n")
+
+				So(output, ShouldNotContainSubstring, "testValue2\n")
+			}
+		})
```

Comment:
Should add a test where the setMetadata := "ibackup:user:testKey1:testValue1Updated"
Actually, that looks really bad. Maybe the metadata that users have to supply should be in the format "key=value", so they can say "ibackup:user:testKey1=testValue1Updated".